include/seqan3/alignment/all.hpp

```diff
@@ -10,20 +10,293 @@
+ *
+ * <small><i>The base type seqan3::scoring_scheme_base is only implementation detail and not required for most users.
+ * Types that model seqan3::ScoringScheme can (but don't need to!) inherit from it.</i></small>
+ *
+ * ### Scoring gaps
+ *
+ * Gaps are usually not scored on a per-character-basis, because it is widely recognised that the likelihood of
+ * `n` consecutive gaps is much higher than that of `n` individual gaps. This is based on the assumption that
+ * one biological event often introduces more than one gap at a time and that single character gaps are not common
+ * due to other biological factors like frame preservation in protein-coding sequences.
+ *
+ * Therefore, the gaps are not handled by the scoring scheme. SeqAn offers the additional seqan3::gap_scheme
+ * which can be used to set the scores for opening or extending a gap.
+ *
+ * The gap scheme can be configured with the seqan3::align_cfg::gap element. If the configuration is not specified,
+ * the algorithm uses edit distance scores for deletion/insertion.
+ *
+ * ## Computing banded alignments
+ *
+ * SeqAn offers the computation of banded alignments to reduce the running time of the algorithm. This can be
+ * helpful if the region in which the optimal alignment exists is known a priori. To specify the banded alignment
+ * the developer can use the seqan3::align_cfg::band option initialised with a seqan3::static_band.
+ * The seqan3::static_band is constructed with a lower bound and an upper bound. The upper bound must always be greater
+ * or equal to the lower bound. To position the band, imagine a rectangle where the first sequence is written on top
+ * and the second sequence along the left vertical side. A negative bound implies a start of the band within the
+ * vertical part while a positive bound implies a start within the top part of this rectangle at the respective
+ * position.
+ *
+ * ## Global and local alignments
+ *
+ * SeqAn also offers the standard global and local alignments.
+ * These configurations can be set via seqan3::align_cfg::mode. The global alignment can be further refined using
+ * the seqan3::align_cfg::sequence_ends configuration. This configuration allows to enable, respectively disable
+ * the scoring of leading and trailing gaps at the respective sequence ends. This option is not available for the
+ * local alignment.
+ *
+ * ## Computing edit distance
+ *
+ * A special implementation of the global sequence alignment is the edit distance. This algorithm computes merely
+ * the number of edit operations (substitutions, insertions and deletions). It can only be computed with a scoring
+ * scheme initialised with `0` for matching letters and `-1` for mismatching letters and a gap scheme with `-1` for
+ * inserting or deleting a gap. Currently, it is only available for nucleotide scoring schemes. There is a special
+ * edit distance config seqan3::align_cfg::edit which is initialised with the correct scoring and gap scheme.
```

Comment:
seqan3::align_cfg::edit is not contained in the table above! Is it compatible with scoring and gap?

Reply:
yes, it is a shortcut for a combinations of other configs. I added the term shortcut to make it more clear.